File: 프로그래머스/2/60057. 문자열 압축/문자열 압축.py
def solution(s):
    answer = len(s)
    
    size = 1
    while(size <= (len(s) // 2)):
        length, count = 0, 0
        pattern = s[0:size]
        
        for i in range(0,len(s),size):
            if s[i:i+size] == pattern:
                count += 1
            else:
                if count == 1:
                    length += len(pattern)
                else:
                    length += len(str(count)) + len(pattern)
                
                pattern = s[i:i+size]
                count = 1
                
        length += len(pattern) if count == 1 else len(str(count)) + len(pattern)
        answer = min(answer, length)
        size += 1
        
    return answer

File: 프로그래머스/2/60057. 문자열 압축/문자열 압축.swift
import Foundation

func solution(_ s: String) -> Int {
    var answer = s.count
    
    var size = 1
    while size <= s.count / 2 {
        var length = 0
        var count = 0
        var pattern = String(s.prefix(size))
        
        var i = 0
        while i < s.count {
            let startIndex = s.index(s.startIndex, offsetBy: i)
            let endIndex = s.index(s.startIndex, offsetBy: min(i + size,s.count))
            let subString = String(s[startIndex..<endIndex])
            
            if subString == pattern {
                count += 1
            } else {
                if count == 1 {
                    length += pattern.count
                } else {
                    length += String(count).count + pattern.count
                }
                
                pattern = subString
                count = 1
            }
            
            i += size
        }
        
        if count == 1 {
            length += pattern.count
        } else {
            length += String(count).count + pattern.count
        }
        
        answer = min(answer, length)
        size += 1
    }
    
    return answer
}